src/transformers/CJSImportTransformer.ts

```diff
@@ -177,20 +177,31 @@ export default class CJSImportTransformer extends Transformer {
     if (!replacement) {
       return false;
     }
-    // We need to change to (0, f) if this is a function call, so that it won't be interpreted as a
-    // method access. This can also happen in situations like (f)(), so a following close-paren
-    // should trigger this behavior as well if it eventually has an open-paren. In some cases, like
-    // export assignees, we must NOT turn the identifier into a normal expression, so we need to
-    // just to the regular replacement.
+    // Tolerate any number of closing parens while looking for an opening paren
+    // that indicates a function call.
     let possibleOpenParenIndex = this.tokens.currentIndex() + 1;
     while (
       possibleOpenParenIndex < this.tokens.tokens.length &&
       this.tokens.tokens[possibleOpenParenIndex].type === tt.parenR
     ) {
       possibleOpenParenIndex++;
     }
+    // Avoid treating imported functions as methods of their `exports` object
+    // by using `(0, f)` when the identifier is in a paren expression. Else
+    // use `Function.prototype.call` when the identifier is a guaranteed
+    // function call. When using `call`, pass undefined as the context.
     if (this.tokens.tokens[possibleOpenParenIndex].type === tt.parenL) {
-      this.tokens.replaceToken(`(0, ${replacement})`);
+      if (this.tokens.tokenAtRelativeIndex(1).type == tt.parenL) {
```

Comment:
`===` please. (This breaks lint, and you should be able to do `yarn test` locally to catch it.)

Also, maybe you could do `this.tokens.currentIndex() + 1 === possibleOpenParenIndex`?

I *think* you should be able to handle the `new` issues by just making sure the prior token is not `new` in this conditional.

Reply:
Sorry, I live on the edge. Will fix it. (regarding `yarn test`: I've been using `jest -i` with a `jest.config.js` I setup, because I can't stand mocha)

> Also, maybe you could do this.tokens.currentIndex() + 1 === possibleOpenParenIndex?

Nice! At this point, we could just merge it and you fix the nit picks. :P